zarr/storage.py

```diff
@@ -1565,7 +1565,7 @@ def __setitem__(self, key, value):
             else:
                 keyinfo.external_attr = 0o644 << 16     # ?rw-r--r--
 
-            self.zf.writestr(keyinfo, value)
+            self.zf.writestr(keyinfo, value.tobytes())
```

Comment:
This would introduce an additional copy before writing to disk.

Guessing the issue is `ZipFile` assumes data here is `uint8` even if that may not be the case. Also suspect this only happens when a compressor is not used (otherwise this would already be the case)

Would suggest changing line 1554 above to add `.view("u1")` to cast to `uint8`. This should have the same end result (data is represented as `uint8`), but not cause a copy

Reply:
Good point. I'll write a test case as well.

Reply:
Test is added now.